mod_modular_vanilla/hooks/skills/skill.nut

```diff
@@ -1,3 +1,38 @@
+::ModularVanilla.QueueBucket.VeryLate.push(function() {
+	::ModularVanilla.MH.hookTree("scripts/skills/skill", function(q) {
+		// This check exists because `skill` does not contain a `create` function.
+		if (q.ClassName != "skill")
+		{
+			// VanillaFix: https://steamcommunity.com/app/365360/discussions/1/604169856012736554/
+			// Vanilla does not update the `m.HitChanceBonus` with respect to the actual MeleeSkill modifier
+			// leading to wrong information in the Hitfactors tooltip.
+			// We set HitChanceBonus always to 0 as base value. Then set it to the correct value at the end
+			// of the onAnySkillUsed function.
+			q.create = @(__original) { function create()
+			{
+				__original();
+				this.m.HitChanceBonus = 0;
+			}}.create;
+
+			q.onAnySkillUsed = @(__original) { function onAnySkillUsed( _skill, _targetEntity, _properties )
+			{
+				if (_skill != this)
+				{
+					__original(_skill, _targetEntity, _properties);
+					return;
+				}
+
+				local mskill = _properties.MeleeSkill;
+				local bonus = this.m.HitChanceBonus;
+
+				__original(_skill, _targetEntity, _properties);
+
+				this.m.HitChanceBonus = bonus - mskill + _properties.MeleeSkill;
```

Comment:
I think this makes the code more readable: 
- marking the variables before the __original call as `old`
- swapping `new` and `old` value in last calculation as it is more intuitive to subtract the old value from the new one in order to get the difference, which we are interested here
```suggestion
				local oldMeleeSkill = _properties.MeleeSkill;
				local oldHitChance = this.m.HitChanceBonus;

				__original(_skill, _targetEntity, _properties);

				this.m.HitChanceBonus = oldHitChance + _properties.MeleeSkill - oldMeleeSkill;
```